src/test/java/api/support/jackson/serializer/JsonObjectJacksonSerializer.java

```diff
@@ -0,0 +1,19 @@
+package api.support.jackson.serializer;
+
+import java.io.IOException;
+
+import com.fasterxml.jackson.core.JsonGenerator;
+import com.fasterxml.jackson.databind.JsonSerializer;
+import com.fasterxml.jackson.databind.SerializerProvider;
+
+import io.vertx.core.json.JsonObject;
+
+/**
+ * Jackson serializer that allows vertx JsonObject to be correctly serialized.
+ */
+public final class JsonObjectJacksonSerializer extends JsonSerializer<JsonObject> {
+  @Override
+  public void serialize(JsonObject value, JsonGenerator gen, SerializerProvider serializers) throws IOException {
+    gen.writeRaw(value.toString());
```

Comment:
Is this intended to write the encoded JSON without modification?

Reply:
When I try to serialize `JsonObject` using jackson's `ObjectMapper`  the resulting json is not correct, the json is:
```
{
  "map": {my-json}
}
```

This serializer resolves this issue.

Reply:
Thanks. I think that answers my question. In effect, this serialiser overrides Jackson's built in serialisation and uses the JsonObject's own JSON representation.

Given that, could this use either `encode()` or `encodePrettily()` to better describe that intent (and be more stable than `toString`)?

Reply:
I use `encode()` now.